.github/workflows/internal-build.yml

```
@@ -33,7 +48,7 @@ jobs:
         run: |
           python3 -u ./tools/cross/internal_build.py \
             ${{github.event.pull_request.number}} \
-            ${{github.event.pull_request.head.sha}} \
+            ${{steps.get_sha.outputs.sha}} \
```

```suggestion
            ${{github.event.pull_request.merge_commit_sha} \
```

The documentation's not the clearest but you may be able to save some work with this: https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request

---

That was my initial attempt, but `merge_commit_sha` was null for some reason.